modules/src/mock_client/state.rs

```diff
@@ -22,7 +22,7 @@ impl MockClientState {
         header: AnyHeader,
     ) -> Result<(MockClientState, MockConsensusState), Box<dyn std::error::Error>> {
         match header {
-            #[cfg(test)]
+            #[cfg(test)] // this is weird...
```

Comment:
Agreed, it looks a bit weird. To clarify just in case, this directive tells the compiler emit this branch when compiling under the test profile, which is required as the `AnyHeader::Mock` variant is similarly only defined under the testing profile.

Reply:
Got it, thanks for the explanation! I had that comment specifically to remember to ask you hehe.